Sources/SwifterSwift/SwiftStdlib/CollectionExtensions.swift

```diff
@@ -22,10 +22,8 @@ public extension Collection {
     ///
     /// - Parameter each: closure to run for each element.
     func forEachInParallel(_ each: (Self.Element) -> Void) {
-        let indicesArray = Array(indices)
-        DispatchQueue.concurrentPerform(iterations: indicesArray.count) { (index) in
-            let elementIndex = indicesArray[index]
-            each(self[elementIndex])
+        DispatchQueue.concurrentPerform(iterations: count) {
```

Comment:
@LucianoPAlmeida Maybe we should be making a copy of `self` first, to make it thread-safe?

Reply:
Maybe, but I think we won't have issues here because as far as I can see it is just read operations on self here. I think that would be the case maybe if this was a mutating function where `self` is implicit `inout` but since it is not I think we wouldn't run into any issues. There is any case you think that maybe problematic?